src/Workspaces/Core/MSBuild.BuildHost/Program.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.Build.Locator;
+using Microsoft.CodeAnalysis.Host.Mef;
+using StreamJsonRpc;
+using System.CommandLine;
+
+#if NETCOREAPP
+using System.Runtime.Loader;
+#endif
+
+namespace Microsoft.CodeAnalysis.Workspaces.MSBuild.BuildHost;
+
+internal static class Program
+{
+    internal static async Task Main(string[] args)
+    {
+        var binaryLogOption = new CliOption<string?>("--binlog") { Required = false };
+        var command = new CliRootCommand { binaryLogOption };
+        var parsedArguments = command.Parse(args);
+        var binaryLogPath = parsedArguments.GetValue(binaryLogOption);
+
+        // Create a MEF container so we can create our SolutionServices to use for discovering language services; we'll include our own assembly in since that contains the services used
+        // for actually loading MSBuild projects.
+        var thisAssembly = typeof(Program).Assembly;
+
+#if NETCOREAPP
+
+        // In the .NET Core case, the dependencies we want to dynamically load are not in our deps.json file, so we won't find them when MefHostServices tries to load them
```

Comment:
which dependencies are these, and why are we not able to include them as regular dependencies?

Reply:
If I recall it was the C# compiler binaries, and I can't recall if we otherwise had problems with double writing them to the same directory or something.

Reply:
I'm now no longer using MEF in this process, so this goes away.